frontend/src/Components/Conditions/ConditionsMap.tsx

```diff
@@ -67,46 +96,78 @@ const getColorForValue = (
   o: number,
 ): string => {
   return value <= g
-    ? '#09BD09' // green
+    ? '#09BD09' // green - Low - value <= g
     : value <= gy
-    ? '#02FC02' // greenyellow
+    ? '#02FC02' // greenyellow - Medium - g < value <= gy
     : value <= y
-    ? '#FFFF00' // yellow
+    ? '#FFFF00' // yellow - High - gy < value <= y
     : value <= o
-    ? '#ff9900' // orange
-    : '#FF0000';
+    ? '#ff9900' // orange - Critical -  y < value
+    : '#FF0000'; //  More Critical
 };
 
 /**
  * The gradient color for each indicator
  *
  * @param properties the type and value of the GeoJSON being drawn
  *
- * @author Kerbourc'h
+ * @author Kerbourc'h, Hansen
  */
 const getConditionColor = (properties: GeoJSON.GeoJsonProperties): string => {
   if (properties !== null) {
     const type = properties.type;
     const value = properties.value;
 
     if (type !== undefined && type !== 'ALL') {
-      switch (type) {
-        case KPI:
-          return getColorForValue(value, 4.0, 6.0, 7.0, 8.0);
-        case DI:
-          return getColorForValue(value, 1.2, 1.5, 2.0, 2.5);
-        case IRI:
-        case IRInew:
-          return getColorForValue(value, 1.5, 1.5, 2.5, 2.5);
-        case Mu:
-          // The minus are a small trick to be able to use the getColorForValue
-          return getColorForValue(-value, -0.8, -0.5, -0.3, -0.2);
-        case Enrg:
-          return getColorForValue(value, 0.05, 0.1, 0.15, 0.25);
-      }
+      const thresholds = getSeverityThresholds(type);
+      getColorForValue(
+        value,
+        thresholds[0],
+        thresholds[1],
+        thresholds[2],
+        thresholds[3],
+      );
     }
   }
-  return 'grey';
+  return '#00000000';
+};
+
+/**
+ * Checking if severity level matches value
+ *
+ * @param severity the selected severity level
+ * @param properties the type and value of the GeoJSON being drawn
+ *
+ * @author Hansen
+ */
+
+const severityThreshold = (
+  severity: SeverityMode,
+  properties: GeoJSON.GeoJsonProperties,
+): boolean => {
+  if (properties !== null) {
+    const type = properties.type;
+    const value: number = properties.value;
+    let thresholds: number[] = [];
+
+    if (type !== undefined && severity.selected) {
+      thresholds = getSeverityThresholds(type);
+
+      const isCritical = severity.mode.includes('Critical');
+      const isHigh = severity.mode.includes('High');
+      const isMedium = severity.mode.includes('Medium');
+      const isLow = severity.mode.includes('Low');
+
+      const meetsSeverityConditions =
+        (isLow && (value <= thresholds[0] || value < thresholds[1])) ||
+        (isMedium && thresholds[1] <= value && value <= thresholds[2]) ||
+        (isHigh && thresholds[2] < value && value <= thresholds[3]) ||
+        (isCritical && thresholds[3] < value);
+
```

Comment:
Currently conditions look like this: 
isLow -> value <= g or value < gy - (green/greenyellow)
isMedium -> gy <= value  <= y
isHigh -> y < value < o
isCritical -> value > o